src/lib.rs

```diff
@@ -118,6 +132,49 @@ pub fn run(config: Config) -> Vec<ViolationWithContext> {
     all_violations.into_values().flatten().collect()
 }
 
+const NUM_INSTANTIATED_PER_FILE_RULES_HASH_BUCKETS: usize = 256;
+
+#[allow(clippy::type_complexity)]
+struct AllInstantiatedPerFileRules(
+    [Mutex<Option<HashMap<PathBuf, HashMap<RuleName, Arc<dyn RuleInstancePerFile>>>>>;
+        NUM_INSTANTIATED_PER_FILE_RULES_HASH_BUCKETS],
+);
+
+impl Default for AllInstantiatedPerFileRules {
+    fn default() -> Self {
+        Self(array::from_fn(|_| Default::default()))
```

Comment:
Weird that it's so hard to initialize a fixed-size array with the default elements?

Reply:
Ah, I've never tried that. It doesn't let you do the usual `[Default::default(); NUM_INSTANTIATED_PER_FILE_RULES_HASH_BUCKETS]` syntax?

Reply:
No and I don't think I really understood why not